api/osf_groups/views.py

```diff
@@ -0,0 +1,224 @@
+from django.apps import apps
+from django.db.models import Q
+
+from rest_framework import generics, permissions as drf_permissions, exceptions
+from rest_framework.exceptions import NotFound, ValidationError
+from guardian.shortcuts import get_perms
+
+from api.base import permissions as base_permissions
+from api.base.exceptions import InvalidFilterOperator, InvalidFilterValue
+from api.base.filters import ListFilterMixin
+from api.base.utils import get_object_or_error, get_user_auth, is_bulk_request
+from api.base.views import JSONAPIBaseView
+from api.base import generic_bulk_views as bulk_views
+from api.base.parsers import JSONAPIMultipleRelationshipsParser, JSONAPIMultipleRelationshipsParserForRegularJSON
+from api.osf_groups.permissions import IsGroupManager, GroupMemberManagement
+from api.osf_groups.serializers import (
+    OSFGroupSerializer,
+    OSFGroupDetailSerializer,
+    OSFGroupMemberSerializer,
+    OSFGroupMemberDetailSerializer,
+    OSFGroupMemberCreateSerializer,
+)
+from api.users.views import UserMixin
+from framework.auth.oauth_scopes import CoreScopes
+from osf.models import OSFGroup, OSFUser
+from osf.utils.permissions import MANAGER, GROUP_MEMBER_PERMISSIONS
+
+
+class OSFGroupMixin(object):
+    """
+    Mixin with convenience method for retrieving the current OSF Group
+    """
+    group_lookup_url_kwarg = 'group_id'
+
+    def get_osf_group(self, check_object_permissions=True):
+
+        group = get_object_or_error(
+            OSFGroup,
+            self.kwargs[self.group_lookup_url_kwarg],
+            self.request,
+            display_name='osf_group',
+        )
+
+        if check_object_permissions:
+            self.check_object_permissions(self.request, group)
+        return group
+
+    def get_node_group_perms(self, group, node):
+        return get_perms(group.member_group, node)
+
+
+class OSFGroupList(JSONAPIBaseView, generics.ListCreateAPIView, ListFilterMixin, OSFGroupMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.OSF_GROUPS_WRITE]
+    model_class = apps.get_model('osf.OSFGroup')
+
+    parser_classes = (JSONAPIMultipleRelationshipsParser, JSONAPIMultipleRelationshipsParserForRegularJSON,)
+    serializer_class = OSFGroupSerializer
+    view_category = 'osf_groups'
+    view_name = 'group-list'
+    ordering = ('-modified', )
+
+    def get_default_queryset(self):
+        return OSFGroup.objects.all()
```

Comment:
Doesn't this need to filter for the groups for the current user?

Reply:
I can add that. There's also a UserGroups endpoint as well.

Reply:
I didn't see anything about groups being public or not in the spec, but I don't see a use case for seeing other people's groups. Better to be conservative in this case, I suppose.

Reply:
Yeah, I think that makes sense.